src/audio/crossover/crossover_generic.c

```diff
@@ -225,32 +225,39 @@ static void crossover_s32_default(struct comp_data *cd,
 				  int32_t num_sinks,
 				  uint32_t frames)
 {
-	struct crossover_state *state;
-	const struct audio_stream *source_stream = bsource->data;
-	struct audio_stream *sink_stream;
-	int32_t *x, *y;
-	int ch, i, j;
-	int idx;
-	int nch = audio_stream_get_channels(source_stream);
-	int32_t out[num_sinks];
+	/* Array to hold active sink streams; initialized to null */
+	struct audio_stream *sink_stream[SOF_CROSSOVER_MAX_STREAMS] = { NULL };
```

Comment:
It would be preferred to have one logic change per git commit. E.g. these comments to existing code are not related to the the optimization, so the two should be in separate git commits for clarity. 

As for the comment, I personally like more comment heavy coding style, but in SOF we traditionally have been more conservative on the amount of comments. I'll let the crossover maintainers chime in here whether they find the comment level ok or not here. Definitely I found these helpful.

Reply:
@johnylin76 and I are the maintainers. I second the atomic commits arguements. Please split "just adding comments" from  your actual change so its easier to understand.

Reply:
Done !! Please take a look.